### PR TITLE
Cgroups

### LINKED ISSUE #29282: [receiver/hostmetrics] Support collecting process cgroup as dimension
### Component(s)

receiver/hostmetrics

### Is your feature request related to a problem? Please describe.

I would like to collect the cgroup associated with a process when collecting process information through the hostmetrics receiver.

### Describe the solution you'd like

The cgroup information should be offered as an additional dimension of the metric related to processes.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Adds `process.cgroup` resource attribute to process metrics

**Link to tracking Issue:**
Fixes #29282

### CODE REVIEW

**Comment by crobert-1** [comment]
[CI test](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7106764063/job/19346884419?pr=29621) is hitting a panic that looks like it's caused by this change:
```
--- FAIL: TestScrapeMetrics_Filtered (0.00s)
    --- FAIL: TestScrapeMetrics_Filtered/No_Filter (0.00s)
panic: interface conversion: processscraper.processHandle is *processscraper.processHandleMock, not *process.Process [recovered]
	panic: interface conversion: processscraper.processHandle is *processscraper.processHandleMock, not *process.Process
```

**Comment by atoulme** [comment]
Yes, I need to work on this some more. The interface used doesn't work well to access fields on the struct.

**Comment by atoulme** [comment]
Sorry, this took a long time. This is now ready for review.

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go
Comment (astencel-sumo) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go:36]: This new configuration option should be added to README.

This would be the fifth "mute" option on the process scraper. Perhaps we should add a global "mute_all_errors" option to make life easier for users? 🤔

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml
Comment (astencel-sumo) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml:50]: We should mention that this is only available on Linux systems, right?

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go
Comment (astencel-sumo) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go:134]: This is not ideal.

I understand that `gopsutil` rejected the proposal. The reason [apparently is](https://github.com/shirou/gopsutil/discussions/1553) that the maintainer of gopsutil wants to introduce a new architecture for platform-specific features in a new major version of the library - v4. Looking at the discussion, the maintainer is a single person working on the library on their weekends. We rely on the library heavily in this component (process scraper), so perhaps it would be nice to contribute to gopsutil, help ship v4 and benefit both projects as a result.

This is not to block this change from merging, but ideally this should be a temporary workaround until gopsutil adds this functionality (with or without our help).

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml
Comment (braydonk) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml:50]: In semantic conventions we are trying to specify system specific metrics under a specific namespace. For example in this scraper, `process.handles` will soon become `process.windows.handles` since it's Windows exclusive. So I imagine we will want to name this `system.linux.cgroup`. It would probably be good to do that here, because that's probably what we'd call it anyway for semconv.

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go
Comment (braydonk) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go:134]: Since `gopsutil` is under a different license, should this code copied from `gopsutil` be moved into another file with their license notifier on top? It is BSD licensed with this clause:
```
 * Redistributions of source code must retain the above copyright notice, this
   list of conditions and the following disclaimer.
```

Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go:134]: @astencel-sumo sure thing, please feel free to offer your help

@braydonk good point, I will take a look

Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go:134]: Added a NOTICE file with mention of copyright and inclusion of gopsutil code. Note I originated the code to gopsutil, so I could make the argument I can lend copyright usage to both projects.

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml
Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml:50]: :shrug: cgroups indeed only exist under Linux. I can add that to the description.

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go
Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go:36]: probably good to look at after this PR lands, if it does

**Review by andrzej-stencel — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go
Comment (braydonk) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/config.go:36]: A solution I'd come up with for this is a new type of error that a scraper can report to log errors at a different level. This combined with option 3 would be a pretty good outcome I think; there would be metrics reported for the different types of errors, and these errors could be logged at debug level by the scraper controller.

This is the issue I have open on the collector repo with accompanying PR: https://github.com/open-telemetry/opentelemetry-collector/issues/8293

**Review by dmitryax — APPROVED** [review]

**Comment by cforce** [comment]
Where to on / se resource attribute process.cgroup: true ?

**Comment by crobert-1** [comment]
@cforce You're wondering how to enable the `process.cgroup` resource attribute? 

The [documentation here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md#resource-attributes) shares a little more information, you'd want add something like this to your `hostmetrics` receiver configuration:

```
resource_attributes:
  process.cgroup:
    enabled: true
```

**Pull request merged**
